bin/waagent2.0

```diff
@@ -3292,7 +3292,7 @@ class Certificates(object):
         index = 1
         filename = str(index) + ".prv"
         while os.path.isfile(filename):
-            pubkey = RunGetOutput(Openssl + " rsa -in " + filename + " -pubout 2> /dev/null ")[1]
+            pubkey = RunGetOutput(Openssl + " pkey -in " + filename + " -pubout 2> /dev/null ")[1]
```

Comment:
I think this is the wrong place to make the change in. We don't use waagent2.0 anymore.

You'll need to make a change in `get_pubkey_from_prv` method of the `CryptUtil` class located in the `azurelinuxagent/common/utils/cryptutil.py` file.

```
    def get_pubkey_from_prv(self, file_name):
        if not os.path.exists(file_name):
            raise IOError(errno.ENOENT, "File not found", file_name)
        else:
            cmd = "{0} rsa -in {1} -pubout 2>/dev/null".format(self.openssl_cmd,
                                                               file_name)
            pub = shellutil.run_get_output(cmd)[1]
            return pub
```

Also, when you make the change in the new location, could you please add a corresponding change in `tests/utils/test_crypt_util.py` as well? 

Thanks

Reply:
ok - I had no idea about walinuxagent not using waagent2.0 anymore. I'm also not a python developer, so it might be better for someone with more python expertise to fix this.

Can you tell me how I can test this on an Azure VM, to make sure it works? [See my comment here, perhaps my workaround for testing this was wrong](https://github.com/Azure/WALinuxAgent/issues/1550#issuecomment-501027678).

Reply:
@vrdmr - I've made the additional change you requested to `cryptutil.py`.

I didn't find any tests in `test_crypt_util.py` referencing `get_pubkey_from_prv()`, so I didn't add a test case for a non RSA cert either. I understand the desirability of adding one, but I'm afraid my attempts at adding a test case would not be a good use of time for either of us. Otoh, this would mostly be testing openssl.

I'd be happy to help test this in Azure, if you can give me guidance on the best way to test this on a VM/vmset.

Reply:
@johncrim - I added couple of unittests. Please let me know if it looks good. Thanks.